packages/react-native/src/core/storage/utils.ts

```diff
@@ -32,3 +34,51 @@ export function prepareGatewayUrls(
 
   return allGatewayUrls;
 }
+
+export function setAnalyticsHeaders(xhr: XMLHttpRequest) {
+  const globals = getAnalyticsGlobals();
+  xhr.setRequestHeader("x-sdk-version", globals.x_sdk_version);
+  xhr.setRequestHeader("x-sdk-os", globals.x_sdk_os);
+  xhr.setRequestHeader("x-sdk-name", globals.x_sdk_name);
+  xhr.setRequestHeader("x-sdk-platform", globals.x_sdk_platform);
+  xhr.setRequestHeader("x-bundle-id", globals.app_bundle_id);
+}
+
+export function getAnalyticsHeaders() {
+  const globals = getAnalyticsGlobals();
+  return {
+    "x-sdk-version": globals.x_sdk_version,
+    "x-sdk-os": globals.x_sdk_os,
+    "x-sdk-name": globals.x_sdk_name,
+    "x-sdk-platform": globals.x_sdk_platform,
+    "x-bundle-id": globals.app_bundle_id,
+  };
+}
+
+export function getAnalyticsGlobals() {
+  if (typeof globalThis === "undefined") {
+    return {
+      x_sdk_name: packageVersion.name,
+      x_sdk_platform: "mobile",
+      x_sdk_version: packageVersion.version,
+      x_sdk_os: Platform.OS,
+      app_bundle_id: appBundleId,
+    };
+  }
+
+  if ((globalThis as any).X_SDK_NAME === undefined) {
+    (globalThis as any).X_SDK_NAME = packageVersion.name;
+    (globalThis as any).X_SDK_PLATFORM = "mobile";
+    (globalThis as any).X_SDK_VERSION = packageVersion.version;
+    (globalThis as any).X_SDK_OS = Platform.OS;
+    (globalThis as any).APP_BUNDLE_ID = appBundleId;
```

Comment:
also don't see where this is called in react-native, was expecting it to get called in ThirdwebProvider